endpoints/data_objects/src/main.cpp

```diff
@@ -353,84 +463,210 @@ namespace
 					}
 				}
 
-				io::client::native_transport tp{conn};
-				io::idstream in{tp, lpath_iter->second};
+				const auto status = fs::client::status(conn, lpath_iter->second);
 
-				if (!in) {
-					log::error("{}: Could not open data object [{}] for read.", fn, lpath_iter->second);
-					res.result(http::status::bad_request);
+				if (!fs::client::is_data_object(status)) {
+					log::error(
+						"{}: Logical path [{}] does not point to a data object or does not exist.",
+						fn,
+						lpath_iter->second);
+					res.result(http::status::internal_server_error);
 					res.prepare_payload();
 					return _sess_ptr->send(std::move(res));
 				}
 
-				auto iter = _args.find("offset");
-				if (iter != std::end(_args)) {
+				std::int64_t offset = 0;
+				if (const auto iter = _args.find("offset"); iter != std::end(_args)) {
 					try {
-						in.seekg(std::stoll(iter->second));
+						offset = std::stoll(iter->second);
 					}
 					catch (const std::exception& e) {
+						log::error("{}: Invalid value for [offset] parameter. Received [{}].", fn, iter->second);
+						res.result(http::status::internal_server_error);
+						res.prepare_payload();
+						return _sess_ptr->send(std::move(res));
+					}
+
+					if (std::cmp_less(offset, 0)) {
 						log::error(
-							"{}: Could not seek to position [{}] in data object [{}].",
+							"{}: Invalid value for [offset] parameter. Must be greater than or equal to 0. Received "
```

Comment:
Any instance of "greater than or equal to 0" could be "non-negative". Might save a line or two. :) No action required
```suggestion
							"{}: Invalid value for [offset] parameter. Must be non-negative. Received "
```

Reply:
Done.